api/tests/unit/features/versioning/test_unit_versioning_views.py

```diff
@@ -638,3 +638,255 @@ def test_disable_v2_versioning_returns_bad_request_if_not_using_v2_versioning(
+        args=[environment_v2_versioning.id, feature.id],
+    )
+
+    data = {
+        "publish_immediately": True,
+        "feature_states_to_update": [
+            {
+                "feature_segment": None,
+                "enabled": True,
+                "feature_state_value": {"type": "unicode", "string_value": "updated!"},
+            }
+        ],
+        "feature_states_to_create": [
+            {
+                "feature_segment": {
+                    "segment": additional_segment_1.id,
+                    "priority": 2,
+                },
+                "enabled": True,
+                "feature_state_value": {
+                    "type": "unicode",
+                    "string_value": "segment-override-1",
+                },
+            },
+            {
+                "feature_segment": {
+                    "segment": additional_segment_2.id,
+                    "priority": 1,
+                },
+                "enabled": True,
+                "feature_state_value": {
+                    "type": "unicode",
+                    "string_value": "segment-override-2",
+                },
+            },
+        ],
+        "segment_ids_to_delete_overrides": [segment.id],
+    }
+
+    # When
+    response = admin_client_new.post(
+        url, data=json.dumps(data), content_type="application/json"
+    )
+
+    # Then
+    assert response.status_code == status.HTTP_201_CREATED
+
+    new_version_uuid = response.json()["uuid"]
+    new_version = EnvironmentFeatureVersion.objects.get(uuid=new_version_uuid)
+
+    assert new_version.feature_states.count() == 3
+
+    new_version_environment_fs = new_version.feature_states.filter(
+        feature_segment__isnull=True
+    ).get()
+    assert new_version_environment_fs.get_feature_state_value() == "updated!"
+    assert new_version_environment_fs.enabled is True
+
+    new_version_segment_fs_1 = new_version.feature_states.filter(
+        feature_segment__segment=additional_segment_1
+    ).get()
+    assert new_version_segment_fs_1.get_feature_state_value() == "segment-override-1"
+    assert new_version_segment_fs_1.enabled is True
+    assert new_version_segment_fs_1.feature_segment.priority == 2
+
+    new_version_segment_fs_2 = new_version.feature_states.filter(
+        feature_segment__segment=additional_segment_2
+    ).get()
+    assert new_version_segment_fs_2.get_feature_state_value() == "segment-override-2"
+    assert new_version_segment_fs_2.enabled is True
+    assert new_version_segment_fs_2.feature_segment.priority == 1
+
+    assert not new_version.feature_states.filter(
+        feature_segment__segment=segment
+    ).exists()
+
+    assert new_version.published is True
+    assert new_version.is_live is True
+
+
+def test_update_and_create_segment_override_in_single_request(
+    feature: Feature,
+    segment: Segment,
+    segment_featurestate: FeatureState,
+    admin_client: APIClient,
```

Comment:
Is this intentional use of `admin_client` instead of `admin_client_new`? Also, many of the tests use the `admin_client_new` which is great, but are we missing out on testing the ACL with `staff_client` too?

Reply:
Good point. I've updated an older test to use staff client and standardised to using `admin_client_new` on these new tests.